### PR TITLE
Move document upsert to collections view, improve user login/logout/sign in methods

### PR DESCRIPTION
**Pull request by mclean-connor**
This PR does a lot but they can be summarized into two key updates:

- moves the document upsert out of the canvas and into collections. this simplifies user interaction so documents/collections are only modified in a single location. all document loaders and associated nodes were removed from nodes whitelist
- modifies the user login/logout/signup experience. users can now login without oauth... this means with normal name/email/password.

A few updates will still need to be made for the login... primarily proper name/email/password handling

### CODE REVIEW

File: packages/ui/src/ui-component/dialog/LoginDialog.js [hunk 1]
```diff
@@ -1,32 +1,60 @@
...
-        type: 'string',
-        placeholder: 'john doe'
+    const nameInput = {
+        label: 'Name',
+        name: 'name',
+        type: 'string'
+    }
+    const emailInput = {
+        label: 'Email',
+        name: 'emaail',
```
Comment (innov8r-44) [inline packages/ui/src/ui-component/dialog/LoginDialog.js:21]: typo `emaail`

File: packages/ui/src/views/apikey/index.js [hunk 2]
```diff
@@ -155,6 +158,18 @@ const APIKey = () => {
         getAllAPIKeysApi.request()
     }
 
+    useEffect(() => {
+        if (getAllAPIKeysApi.error) {
+            if (getAllAPIKeysApi.error?.response?.status === 401) {
+                setLoginDialogProps({
```
Comment (innov8r-44) [inline packages/ui/src/views/apikey/index.js:164]: if it isn't a 401 should we add a default error dialog as well?

File: packages/ui/src/views/collections/index.js [hunk 3]
```diff
@@ -118,6 +117,24 @@ const Collections = () => {
...
+    }
+
+    useEffect(() => {
+        if (getAllCollectionsApi.error) {
+            if (getAllCollectionsApi.error?.response?.status === 401) {
+                setLoginDialogProps({
+                    title: 'Login',
+                    confirmButtonName: 'Login'
+                })
+                setLoginDialogOpen(true)
```
Comment (innov8r-44) [inline packages/ui/src/views/collections/index.js:133]: Similar here, though i am not sure we would expect many other errors other than authentication issues so maybe not.

File: packages/ui/src/views/credentials/index.js [hunk 4]
```diff
@@ -145,6 +148,18 @@ const Credentials = () => {
...
+    useEffect(() => {
+        if (getAllCredentialsApi.error) {
+            if (getAllCredentialsApi.error?.response?.status === 401) {
+                setLoginDialogProps({
+                    title: 'Login',
+                    confirmButtonName: 'Login'
+                })
+                setLoginDialogOpen(true)
+            }
+        }
```
Comment (innov8r-44) [inline packages/ui/src/views/credentials/index.js:160]: could we extract this function, it looks like very similar code repeated 3x, we could put it into a `loginCheck` (or a better name) and store it in a common area.

**Review by innov8r-44 — COMMENTED** [review]
Some minor changes or improvements, I wasn't able to test yet due to local set up issues. Good improvements!

File: packages/ui/src/views/credentials/index.js (hunk 4 above)
Comment (innov8r-44) [reply packages/ui/src/views/credentials/index.js:160]: Maybe something like this:

```
const checkApiErrorAndHandleLogin = apiResponse => 
    apiResponse.error?.response?.status === 401 ? (
        setLoginDialogProps({
            title: 'Login',
            confirmButtonName: 'Login'
        }),
        setLoginDialogOpen(true)
    ) : null;
  
    ```

File: packages/ui/src/views/apikey/index.js (hunk 2 above)
Comment (mclean-connor) [reply packages/ui/src/views/apikey/index.js:164]: going to defer alternate error handling to separate PR. I can add a Trello ticket for modifying the login dialog. we will have to update how dynamic elements are added / removed.

File: packages/ui/src/views/credentials/index.js (hunk 4 above)
Comment (mclean-connor) [reply packages/ui/src/views/credentials/index.js:160]: adding this to packages/ui/src/utils/genericHelper.js

**Pull request merged**